File: typescript/src/events/topics.ts
/**
 * Kafka topic names for event routing
 * Converted from go/events/topics.go
 */

export const KafkaTopics = {
  ASSET_UPDATES: 'asset-updates',
  EMERGENCY_NOTIFICATIONS: 'emergency-notifications',
  CHAT_MESSAGES: 'chat-messages',
  TACTICAL_COMMANDS: 'tactical-commands',
  LOCATION_UPDATES: 'location-updates',
  VITALS_UPDATES: 'vitals-updates',
  SYSTEM_STATUS: 'system-status',
  VIDEO_UPLOADS: 'video-uploads',
  VIDEO_PROCESSING: 'video-processing',
  FRAME_EXTRACTION: 'frame-extraction',
  FRAME_UPLOAD_COMPLETE: 'frame-upload-complete',
  AI_ANALYSIS: 'ai-analysis',
  EVENT_ANALYSIS: 'event-analysis',
  CAMERA_EVENTS: 'camera-events',
  SUGGESTIONS: 'suggestions',
  MISSION_EVENTS: 'mission-events',
  AI_MISSION_SUGGESTIONS: 'ai-mission-suggestions',
  MISSION_CHAT: 'mission-chat',
} as const;

export type KafkaTopicName = typeof KafkaTopics[keyof typeof KafkaTopics];

File: go/events/topics.go
package events

// Kafka topic names for event routing
// Topics are managed by Kafka admin and must be pre-created in production

// KafkaTopics defines the Kafka topics used by the system
var KafkaTopics = struct {
	AssetUpdates           string
	EmergencyNotifications string
	ChatMessages           string
	TacticalCommands       string
	LocationUpdates        string
	VitalsUpdates          string
	SystemStatus           string
	VideoUploads           string
	VideoProcessing        string
	FrameExtraction        string
	FrameUploadComplete    string
	AIAnalysis             string
	EventAnalysis          string
	CameraEvents           string
	Suggestions            string
	MissionEvents          string
	AIMissionSuggestions   string
	MissionChat            string
}{
	AssetUpdates:           "asset-updates",
	EmergencyNotifications: "emergency-notifications",
	ChatMessages:           "chat-messages",
	TacticalCommands:       "tactical-commands",
	LocationUpdates:        "location-updates",
	VitalsUpdates:          "vitals-updates",
	SystemStatus:           "system-status",
	VideoUploads:           "video-uploads",
	VideoProcessing:        "video-processing",
	FrameExtraction:        "frame-extraction",
	FrameUploadComplete:    "frame-upload-complete",
	AIAnalysis:             "ai-analysis",
	EventAnalysis:          "event-analysis",
	CameraEvents:           "camera-events",
	Suggestions:            "suggestions",
	MissionEvents:          "mission-events",
	AIMissionSuggestions:   "ai-mission-suggestions",
	MissionChat:            "mission-chat",
}
